raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansGroupingTransformer.java

```diff
@@ -71,9 +95,29 @@ public void init(ProcessorContext context) {
   public KeyValue<String, StructuredTrace> transform(TraceIdentity key, RawSpan value) {
     ValueAndTimestamp<RawSpans> rawSpans = inflightTraceStore.get(key);
     RawSpans agg = rawSpans != null ? rawSpans.value() : RawSpans.newBuilder().build();
+    if (maxSpanCountMap.containsKey(key.getTenantId()) && agg.getRawSpans().size() > maxSpanCountMap.get(key.getTenantId())) {
+      if (logger.isDebugEnabled()) {
+        logger.debug("Number of spans in tenant_id={}, trace_id={} is {}",
+          key.getTenantId(),
+          HexUtils.getHex(key.getTraceId()),
+          agg.getRawSpans().size());
+        logger.debug("Dropping span [{}] from tenant_id={}, trace_id={}", value, key.getTenantId(), HexUtils.getHex(key.getTraceId()));
+      }
+      // increment the counter for dropped spans
+      droppedSpansCounter.computeIfAbsent(key.getTenantId(),
+        k -> PlatformMetricsRegistry.registerCounter(DROPPED_SPANS_COUNTER, Map.of("tenantId", k))).increment();
+      return null;
```

Comment:
An observation - since we don't update the last span received timestamp it is possible that new trace with same trace_id gets formed.

Reply:
Yes - once it hits the max span limit, a structured trace will be emitted after 30 seconds. After that, a new trace with same trace_id will be created.